src/bt_peer_connection.cpp

```diff
@@ -121,6 +121,35 @@ namespace {
 } // anonymous namespace
 #endif
 
+bool validate_hash_request(hash_request const& hr, file_storage const& fs)
```

Comment:
this is the key part of this patch. This is according to my understanding of what those fields are. Please correct me if I'm wrong!